utilities/include/vec4.hpp

```diff
@@ -0,0 +1,149 @@
+#ifndef VEC4_HPP
+#define VEC4_HPP
+
+#include <ostream>
+#include <cmath>
+#include <xsimd/xsimd.hpp>
+
+#define ALPHA_ 0 //include alpha or not.
+
+#if XSIMD_WITH_AVX512BW || XSIMD_WITH_AVX512DQ || XSIMD_WITH_AVX512F
+    #define ALIGN_WIDTH 64
+#elif XSIMD_WITH_FMA3_AVX2 || XSIMD_WITH_AVX2 || XSIMD_WITH_FMA3_AVX || XSIMD_WITH_AVX
+    #define ALIGN_WIDTH 32
+#elif XSIMD_WITH_SSE4_2 || XSIMD_WITH_SSE4_1 || XSIMD_WITH_SSSE3 || XSIMD_WITH_SSE3 || XSIMD_WITH_SSE2
+    #define ALIGN_WIDTH 16
+#endif
+
+template<class T, class A = xsimd::default_arch>
+class alignas(ALIGN_WIDTH) vec4{
+    public:
+        constexpr vec4() : m_v{0,0,0,0} {}
+        constexpr vec4(T x) : m_v{x,x,x,x} {}
```

Comment:
It is a good idea to mark single argument constructors as `explicit`, to prevent nasty implicit conversions.

Reply:
will do